pkg/agent/plugin/workloadattestor/unix/unix.go

```diff
@@ -49,13 +53,46 @@ type PSProcessInfo struct {
 }
 
 func (ps PSProcessInfo) NamespacedExe() string {
-	procPath := os.Getenv("HOST_PROC")
-	if procPath == "" {
-		procPath = "/proc"
-	}
+	procPath := getProcPath()
 	return filepath.Join(procPath, strconv.Itoa(int(ps.Pid)), "exe")
 }
 
+// Groups returns the supplementary group IDs
+// This is a custom implementation that only works for linux until the next issue is fixed
+// https://github.com/shirou/gopsutil/issues/913
+func (ps PSProcessInfo) Groups() ([]string, error) {
+	if runtime.GOOS != "linux" {
+		return []string{}, nil
+	}
+
+	procPath := getProcPath()
+	pid := strconv.FormatInt(int64(ps.Pid), 10)
```

Comment:
nit: now this function and NamespacedExe have a slightly different way of forming the proc path to a given pid. Maybe it's worth have a function like `getProcPath(pid int, path ...string)` that can be called like

```
getProcPath(ps.Pid, "status")
```
or
```
getProcPath(ps.Pid, "exe")
```

Reply:
💯